internal/content_coding.go

```diff
@@ -370,6 +395,23 @@ func (d *ZlibDecoder) Decode(data []byte, maxDecompressionSize int64) ([]byte, e
 	return d.buf.Bytes(), nil
 }
 
+type ZstdDecoder struct {
+	decoder *zstd.Decoder
+}
+
+func NewZstdDecoder() (*ZstdDecoder, error) {
+	d, err := zstd.NewReader(nil, zstd.WithDecoderConcurrency(0))
+	return &ZstdDecoder{
+		decoder: d,
+	}, err
+}
+
+func (*ZstdDecoder) SetEncoding(string) {}
+
+func (d *ZstdDecoder) Decode(data []byte, _ int64) ([]byte, error) {
+	return d.decoder.DecodeAll(data, nil)
+}
```

Comment:
I can't do it and I think nobody is using it as well. The reason is that the library gives this option when creating the encoder not when decoding.

Reply:
I see the others return an error when the decoded data is larger in size.

Reply:
I'll leave this to @powersj or @srebhan to decide if it's needed or not.

Reply:
https://pkg.go.dev/github.com/klauspost/compress@v1.16.5/zstd#WithDecodeAllCapLimit

It's disabled by default in zstd. I don't know if it makes sense to put a limit here though. As I said it's not used rn anywhere and perhaps could be improved in the future if it doesn't work fine.

Reply:
I think moving the maximum buffer size to the creation is now easy with the new options interface. Would be happy to see it for this decoder and then also for the others (as a follow-up PR).

Reply:
I don't know if it will be easy because it will confuse the user. This option won't be used by deflate packages (gzip and zlib) but it will be used by zstd at the same time the decode function has to have the maxDecompresionSize anyway for deflate packages as it's only available there.

Reply:
I think @srebhan means to have this as a new config option like we have `encoderConfig` for encoders..